test/Microsoft.Extensions.Caching.Memory.Tests/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -368,6 +369,39 @@ public void SetGetAndRemoveWorksWithObjectKeysWhenDifferentReferences()
             Assert.Null(result);
         }
 
+        [Fact]
+        public async Task GetAndSet_AreThreadSafe()
+        {
+            var cache = CreateCache();
+            string key = "myKey";
+            var cts = new CancellationTokenSource();
+            var cts2 = new CancellationTokenSource();
+
+            cache.Set(key, new Guid(), new MemoryCacheEntryOptions().AddExpirationToken(new CancellationChangeToken(cts.Token)));
+
+            var task1 = Task.Run(() =>
+            {
+                while (!cts2.IsCancellationRequested)
+                {
+                    cache.Set(key, new Guid());
+                }
+            });
+
+            var task2 = Task.Run(() =>
+            {
+                while (!cts2.IsCancellationRequested)
+                {
+                    cache.Get(key);
```

Comment:
This verifies that the original issue was fixed (i.e. the IOE is no longer thrown), but it would also be nice to verify that one thread doesn't affect the others view of the `Scopes`.